CONFIGURATION.md

```diff
@@ -1,79 +1,171 @@
+- [Environment Variables](#environment-variables)
+- [Configuration Files](#configuration-files)
+- [Programmatic Configuration](#programmatic-configuration)
+- [Advanced Configuration](#advanced-configuration)
+- [Configuration Reference](#configuration-reference)
+- [Troubleshooting](#troubleshooting)
+
+## Quick Start
+
+To get started quickly, you only need to set your service key:
 
 ```bash
-export SW_APM_SERVICE_KEY=<set-service-key-here>
+export SW_APM_SERVICE_KEY=<your-api-token>:<your-service-name>
 ```
 
-Configuration can be set several ways, with the following precedence:
+By default, all applicable instrumentations are enabled and the gem works out-of-the-box with sensible defaults.
+
+### Minimal Example
+
+```ruby
+# Set the service key (required)
+ENV['SW_APM_SERVICE_KEY'] = 'your-api-token:my-ruby-app'
+
+# Require the gem (typically done automatically by Bundler)
+require 'solarwinds_apm'
+
+# Your application code here
+```
+
+## Configuration Precedence
+
+Configuration can be set in multiple ways with the following precedence order (highest to lowest):
+
+1. **Environment Variables** - Highest priority
+2. **Programmatic Configuration** - Set in Ruby code
+3. **Configuration Files** - Rails initializer or config file
+4. **Default Values** - Built-in defaults
+
+> **💡 Tip:** Environment variables always take precedence, making them ideal for deployment-specific settings.
 
-`environment variable > programmatic > configuration file > default`
+Environment variables are the most flexible way to configure the SolarWinds APM gem, especially in containerized or cloud environments.
 
-## Environment Variables
+All SolarWinds APM-specific settings are prefixed with `SW_APM_`. Standard OpenTelemetry environment variables are also supported where applicable.
 
-Settings specific to `solarwinds_apm` are prefixed by `SW_APM_` and described in the [Reference](#reference) section. Standard OpenTelemetry environment variables that impact this library's functionality are noted below.
+**Required Configuration**
 
-### Exporter
+| Variable | Description | Example |
+|----------|-------------|---------|
+| `SW_APM_SERVICE_KEY` | API token and service name (required) | `your-token:my-service` |
 
-The default `solarwinds` exporter which communicates with the SolarWinds Observability backend is always configured as OTLP Exporter. Additional exporters can be configured via the `OTEL_TRACES_EXPORTER` environment variable. For example, console exporter is part of standard installation and can be enabled via:
+**Common Configuration**
+
+| Variable | Description | Default | Example |
+|----------|-------------|---------|---------|
+| `SW_APM_ENABLED` | Enable/disable the entire library | `true` | `false` |
+| `SW_APM_DEBUG_LEVEL` | Logging verbosity (-1 to 6) | `3` | `5` |
+| `SW_APM_COLLECTOR` | Collector endpoint override | `apm.collector.na-01.cloud.solarwinds.com:443` | `custom.collector.com:443` |
+
```

Comment:
Could maybe add a mention here of the `## Configuration Reference` below, but up to you.

Reply:
Referenced the `## Configuration Reference`